apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -1073,11 +1073,33 @@ function handleSiteEditorBackButton( calypsoPort ) {
 	// have to do this event delegation style because the Editor isn't fully initialized yet.
 	document.getElementById( 'wpwrap' ).addEventListener( 'click', ( event ) => {
 		const clickedElement = event.target;
-		if ( clickedElement.classList.contains( 'edit-site-navigation-panel__back-to-dashboard' ) ) {
+		const isOldDashboardButton =
+			clickedElement.classList.contains( 'edit-site-navigation-panel__back-to-dashboard' ) ||
+			// The above fails if user clicked internal SVG. So check the parent again.
+			clickedElement.parentElement?.classList.contains(
+				'edit-site-navigation-panel__back-to-dashboard'
+			) ||
+			// The above fails if the user clicked the internal path. So check the parent's parent...
+			clickedElement.parentElement?.parentElement?.classList.contains(
+				'edit-site-navigation-panel__back-to-dashboard'
+			);
```

Comment:
Yes, this is a bit ridiculous. This is related to extending the old handling.

In testing this PR I found the old handling was lacking, that is if a user clicked the `<` svg or path elements instead of the actual dashboard button this would not work and the user would be trapped in the iframe.

This is a quick fix for that gap, which will hopefully no longer be relevant after the next Gutenberg release anyways. (or once we get rid of the calypso iframe, which we should aim to do soon!)